### DISCUSSION TITLE
Gradient is not working on MCU

### DISCUSSION THREAD

**Discussion by yanshay** [slint-ui/slint#4388] [Q&A] [answered]
I'm using ESP32S3 and gradients don't show. The same gradient (though in a much simpler aplication) in the SlintPad shows just fine.
Also in the VSCode Live preview of my app it doesn't show.
I only see the first color of the gradient on the entire rectangle. 
Is it a limitation of the software renderer? Or maybe something with my app which could disable it?

**Comment by ogoffart** [comment]
Gradiants works, but not on the Window and not if the rectangle has border radius.
This is a known limitation currently: https://slint.dev/snapshots/maste/docs/slint/src/advanced/backends_and_renderers
Issue: https://github.com/slint-ui/slint/issues/4176

**Comment by yanshay** [comment]
It's a rectangle, and has no border-radius (I saw that limitation so removed it).

It's basically the PushButton from your printer demo that I adapted, it inherits from a Rectangle and not a rectangle - maybe that's the issue?

The only "special" things about it is:
1. The color comes from the Palette structure and not directly in the button's code.
2. The color is based on a calculation, where the selected color is the gradient
3. It contains a Touch area and a label inside

Could any of these be the cause?

I'll try to troubleshoot further if it is supposed to work.

**Comment by yanshay** [comment, marked as answer]
Ok, found the issue. It's caused by (1) above.
I put the gradient in a property of type color, but gradient is really a brush.
The compiler accepted assigning the gradient into the color, but just picked the first color (would you consider it a bug?).

I also see it doesn't work on round border while on slintpad it does, I guess SlintPad uses a different renderer?

Speaking of round corners - something I noticed, when I set the color of a border and use border-radius, some of the pixels on/around the corners are of different color. So when I placed a round rectangle on another larger one with the same color, around the corners I was able to notice the round corners of the smaller rectangle while everything was a single color. Not a big deal for me, but just mentioning it while on round corners in case someone encounters it.

**Comment by jianjiabailu** [comment]
Can you tell me what screen you're using?  I'm using esp32s3 + gc9503 （esp32_s3_lcd_ev_board）, but my screen doesn't show anything!

**Reply by yanshay** [reply]
I was using WT32-SC01 Plus here.
